Feature/interoperability outside germany screen (EXPOSUREAPP-2264)

BMItr opened the pull request:
[Android, Interop] Screen: Target > Germany (traveling outside of Germany?)

## Relates to
https://jira.itc.sap.com/browse/EXPOSUREAPP-1975  (Overall)
https://jira.itc.sap.com/browse/EXPOSUREAPP-2264 (this Subtask)

## Description
Contains Fragment, Layouts, Viewmodel for corresponding task (Traveling outside of Germany -> Ja, Nein, Keine Angabe)

## Quicktest
Quicktest is available in
Commit: e800f98be88f704e605db804bea3a189392b59de [e800f98]
Just checkout this commit, build it, click on the menu button and select the interop fragment you would like to see

## Note
Keep in mind, that all interop subtasks / screens are devloped independently. The "Marriage" is done seperatly by followup task.

kolyaopahle committed:
Added first set of country flags

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Added first implementation of Submission Country Selection Screen. (Currently not part of nav graph).
Moved stuff inside of submission package into subpackages

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Added missing changes to nav_graph file

apopovsap committed:
Merge pull request #1100 from corona-warn-app/feature/interoperability-country-screen

Initial implementation of submission country selection ui (EXPOSUREAPP-2265)

mertsafter committed:
Test Menu options for Country selection and Measurement (EXPOSUREAPP-2229) (#1086)

* Implement country filter (#2229)

* Implement measure for Risk calculation and Key Retrieval (#2229)

* Some code cleanup for measureRiskLevelAndKeyRetrieval

* Some code cleanup (#2229)

* Implement usage of AppConfig usage for country codes (#2229)

* change CURRENT_COUNTRY val to var

* Suppress "LongMethod" warning (#2229)

* Adjust unit tests to respect new country filter

* Align with backend for app config country property name

* Display total file size of keys in results (#2229)

* Fix linting issue

* Update strings.xml files

* Do refactoring and some code clean up

* Move RiskLevel and Key retrieval measurement in own class

* Move code in logical classes. Remove translation in test fragment

* Fix linting and Unit tests

* Remove 3hour boolean check from debug class.

* Refactor CountryWrapper and missing dates check.

* Fix linting and Unit tests

* Add Unit test for getMissingDates

* Code cleanup

* Hide keyboard on action

* Extend statistics for api submission

* Set callback to null after invoked

Co-authored-by: Mert Safter <mert.safter@sap.com>
Co-authored-by: Matthias Urhahn <matthias.urhahn@sap.com>

d4rken committed:
Merge branch 'dev' into feature/interoperability

# Conflicts:
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/transaction/RetrieveDiagnosisKeysTransaction.kt
#	Corona-Warn-App/src/test/java/de/rki/coronawarnapp/transaction/RetrieveDiagnosisKeysTransactionTest.kt

kolyaopahle committed:
Added initial set of viewModel tests

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Merge branch 'feature/interoperability' into feature/interoperability-country-screen

BMItr committed:
interop - outOfGermany BasicPattern (WIP)

BMItr committed:
adjusted viewmodel, fragments, layouts, includes, strings (WIP)

BMItr committed:
event and binding structure completed

BMItr committed:
corrected button handling, highlight, layout, cleanup

BMItr committed:
lint clean, code reformat

BMItr committed:
improved viewmodel

BMItr committed:
always use viewLifecycleOwner in Framents

BMItr committed:
added companion obj to bottom according to style guide - quick fragment test possible for pr inside the menu

BMItr committed:
cleaned toDos and quickTest, reformat

MertSafter commented:
LGTM

mertsafter reviewed (commented):
LGTM

BMItr committed:
integrated basic outofgermany tests, not much to test here, klint clean

BMItr committed:
prevent merge conflict by countrySelectViewModelTest Update

mertsafter reviewed (dismissed):
👍

d4rken commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/submission/viewmodel/SubmissionOutOfGermanyViewModelTest.kt:
```diff
@@ -0,0 +1,58 @@
+    fun textButtonClickedNone() {
+        submissionOutOfGermanyViewModel.buttonClicked(ButtonClicked.NONE)
+
+        submissionOutOfGermanyViewModel.yesAnswerActive.value?.let { assertFalse(it) }
```
`?` to `!!` so tests fail on null.

Same for `?` below :point_down:

BMItter replied:
I wanted to change it to !! but I discovered a bigger issue behind regarding transformations which took me some brain. I rewrote it and its now safe and will fail on null.

BMItr committed:
updated SubmissionOutOfGermanyViewModel-Test

Transformations in Viewmodel had no observer what avoids quick solution as before

BMItr committed:
Better SubmissionOutOfGermanyViewModel testing

BMItter commented:
Due Storychange this PR is obsolet. Very low probability to come back ever.
